src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/routing/AuditMessageRouter.java

```diff
@@ -65,22 +63,18 @@ public AuditMessageRouter(final Settings settings, final Client clientProvider,
 		}
 	}
 
-	public void setComplianceConfig(ComplianceConfig complianceConfig) {
-		this.complianceConfig = complianceConfig;
-	}
-
 	public boolean isEnabled() {
 		return this.enabled;
 	}
 
-	public final void route(final AuditMessage msg) {
```

Comment:
keep `public final void route(final AuditMessage msg)` and use the most common default to call `route(msg, ...)`

Reply:
What do you mean by most common default ?

Reply:
```
route(msg, true)
```

Reply:
Are you saying to have two functions
1) route(msg, true)
2) route(msg) calling the above ? 

route(msg) would probably be useful only for tests (which I have updated already). Dont want to add more code to main just for tests.

Reply:
It is OK to provide a default so that test don't need to be changed.

Reply:
The tests are already updated

Reply:
Please consider the following approach:
1. revert changes to the `route()` method (it should only take `AuditMessage msg` as an argument)
2. introduce a `boolean` instance variable `areRoutesEnabled` (`false` by default).
3. in `route()` instead of  `isComplianceConfigEnabled` use `areRoutesEnabled`.
4. rename `setupRoutes()` to `enableRoutes()` and make it public
5. do not call `enableRoutes()` from `AuditMessageRouter` constructor, instead, call it from `AuditLogImpl` if it is `enabled` and `complianceConfig` is also enabled

Reply:
Could you explain the value in doing this and insisting on removing the current change ? 

IMO, I see no issue in route(msg, isComplianceConfigEnabled) as it determines what route to take from the audit log based on current compliance configuration for hot reload. 

Also would appreciate if you could send the changes in a PR against this branch so that we can quickly move forward :)

Reply:
The value of the proposal is that
1. `areRoutesEnabled` is the state of `AuditMessageRouter` instance that is simply a delegate of `AuditLogImpl`, not a parameter for the `route()` method (that should only take message to route). Such state will be shared by many calls to the `route()` until there is a state transition.
2. It is not necessary to initialize routes if routes are not enabled.

Reply:
Please see https://github.com/sujithvm/security/pull/57

Reply:
I would prefer AuditMessageRouter improvements to be done separately.
Would appreciate if you could share your changes :) Dont want to trigger another set of changes around this.

Reply:
The improvements to `AuditMessageRouter` are minimal, but it is necessary to keep original `route()` to take message as the only parameter.